website/project/model.py

```diff
@@ -281,6 +283,72 @@ def node(self):
         return self.node__keyed[0] if self.node__keyed else None
 
 
+class ApiOAuth2Application(StoredObject):
+    """Registration and key for user-created OAuth API applications"""
+
+    # Client ID
+    _id = fields.StringField(
+        primary=True,
+        default=lambda: str(ObjectId())
+    )
+
+    # Client ID and secret
+    client_id = fields.StringField(default=lambda: uuid.uuid4().hex,
```

Comment:
Does modularodm provide a way to enforce uniqueness on a field? (all client IDs should be unique)